docs/design/security.md

```diff
@@ -1,17 +1,106 @@
+  * decide which pods should be share files with other pods, and which should not.
+  * reason about application level security, such as containing the effects of a local-file-read exploit in a webserver pod.
+  * do not often reason about operating system or organizational security.
+  * are not necessarily comfortable reasoning about the security properties of a system at the level of detail of Linux Capabilities, SELinux, AppArmor, etc.
+
+* Project Admins:
+  * allocate identity and roles within a namespace
+  * reason about organizational security within a namespace
+    * don't give a developer permissions that are not needed for role.
+    * protect files on shared storage from unnecessary cross-team access
+  * are less focused about application security
+
+* Administrators:
+  * are less focused on application security. Focused on operation system security.
+  * protect the node from bad actors in containers, and properly-configured innocent containers from bad actors in other containers.
+  * comfortable reasoning about the security properties of a system at the level of detail of Linux Capabilities, SELinux, AppArmor, etc.
+  * decides who can use which Linux Capabilities, run privileged containers, use hostDir, etc.
+    * e.g. a team that manages Ceph or a mysql server might be trusted to have raw access to storage devices in some organizations, but teams that develop the applications at higher layers would not.
+
+
+## Proposed Design
+
+A pod runs in a *security context* under a *service account* that is defined by an administrator or project administrator, and the *secrets* a pod has access to is limited by that *service account*.
+
+
+1. The API should authenticate and authorize user actions [authn and authz](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/access.md)
+2. All infrastructure components (kubelets, kube-proxies, controllers, scheduler) should have an infrastructure user that they can authenticate with and be authorized to perform only the functions they require against the API.
+3. Most infrastructure components should use the API as a way of exchanging data and changing the system, and only the API should have access to the underlying data store (etcd)
+4. When containers run on the cluster and need to talk to other containers or the API server, they should be identified and authorized clearly as an autonomous process via a [service account](https://github.com/GoogleCloudPlatform/kubernetes/pull/2297)
+   a.  If the user who started a long-lived process is removed from access to the cluster, the process should be able to continue without interruption
+   b.  If the user who started processes are removed from the cluster, administrators may wish to terminate their processes in bulk
+   c.  When containers run with a service account, the user that created / triggered the service account behavior must be associated to the container's action
```

Comment:
I think it's a timeframe thing.  Ultimately someone needs to correlate ownership of resources to organizational changes, so I'd want to have it called out as a scenario somehow.